containers/src/Kokkos_Vector.hpp

```diff
@@ -147,6 +147,85 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
 
   void clear() { _size = 0; }
 
+  iterator insert(iterator it, const value_type& val) {
+    return insert(it, 1, val);
+  }
+
+  iterator insert(iterator it, size_type count, const value_type& val) {
+    if ((_size == 0) && (it == begin())) {
```

Comment:
This should actually just be `if(it == end())`, because `resize` will work no matter what in that case.

Reply:
No if the vector was not empty, `resize(n,val)` will fill the entire vector with `val` not just the `n` new values.